crates/multihash/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+use multihash::{Code, MultihashRef};
+use std::{convert::TryFrom, io::Write};
+
+use rustler::{Binary, Encoder, Env, NifResult, OwnedBinary, Term};
+
+rustler::atoms! {
+    ok,
+    error,
+    invalid_code,
+}
+
+#[rustler::nif(schedule = "DirtyCpu")]
```

Comment:
Even on the slowest of systems, I can't imaging this will need to marked as `DirtyCpu`.

Here's an experiment:

```diff
diff --git a/crates/multihash/src/lib.rs b/crates/multihash/src/lib.rs
index a1c49f6..682347f 100644
--- a/crates/multihash/src/lib.rs
+++ b/crates/multihash/src/lib.rs
@@ -9,8 +9,23 @@ rustler::atoms! {
     invalid_code,
 }
 
-#[rustler::nif(schedule = "DirtyCpu")]
+struct Timer(::std::time::Instant);
+
+impl Timer {
+    fn new() -> Self {
+        Self(::std::time::Instant::now())
+    }
+}
+
+impl Drop for Timer {
+    fn drop(&mut self) {
+        eprintln!("elapsed {:?}", self.0.elapsed())
+    }
+}
+
+#[rustler::nif]
 fn digest<'a>(env: Env<'a>, data: Binary, raw_code: u64) -> NifResult<Term<'a>> {
+    let _ = Timer::new();
     match Code::try_from(raw_code) {
         Ok(digest_code) => {
             let digest = digest_code.digest(data.as_slice());
@@ -24,6 +39,7 @@ fn digest<'a>(env: Env<'a>, data: Binary, raw_code: u64) -> NifResult<Term<'a>>
 
 #[rustler::nif]
 fn code<'a>(env: Env<'a>, data: Binary) -> NifResult<Term<'a>> {
+    Timer::new();
     match MultihashRef::from_slice(data.as_slice()) {
         Ok(hash) => Ok((ok(), u64::from(hash.algorithm())).encode(env)),
         Err(_) => Ok((error(), invalid_code()).encode(env)),
```

Output: 

```
$ rebar3 eunit
===> Verifying dependencies...
    Finished release [optimized] target(s) in 0.02s
===> Analyzing applications...
===> Compiling multihash
===> Performing EUnit tests...
elapsed 104ns
elapsed 23ns
elapsed 23ns
elapsed 24ns
elapsed 18ns
elapsed 23ns
elapsed 17ns
elapsed 22ns
elapsed 17ns
elapsed 24ns
elapsed 17ns
elapsed 24ns
elapsed 18ns
elapsed 23ns
elapsed 17ns
elapsed 23ns
elapsed 17ns
elapsed 24ns
elapsed 18ns
elapsed 22ns
elapsed 18ns
elapsed 24ns
elapsed 17ns
elapsed 24ns
elapsed 17ns
elapsed 24ns
elapsed 18ns
elapsed 24ns
elapsed 19ns
elapsed 24ns
elapsed 18ns
elapsed 24ns
elapsed 26ns
elapsed 26ns
elapsed 17ns
..
Finished in 0.055 seconds
2 tests, 0 failures
```